Shared/Samples/Apply mosaic rule to rasters/ApplyMosaicRuleToRastersView.swift

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ApplyMosaicRuleToRastersView: View {
+    /// The current draw status of the map.
+    @State private var currentDrawStatus: DrawStatus = .inProgress
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The data model for the sample.
+    @StateObject private var model = Model()
+    
+    /// Holds the reference to the currently selected rule.
+    @State private var ruleSelection: RuleSelection = .objectID
+    
+    var body: some View {
+        MapViewReader { mapProxy in
+            MapView(map: model.map)
+                .onDrawStatusChanged { drawStatus in
+                    // Updates the state when the map's draw status changes.
+                    withAnimation {
+                        currentDrawStatus = drawStatus
+                    }
+                }
+                .overlay(alignment: .center) {
+                    if currentDrawStatus == .inProgress {
+                        ProgressView("Loading...")
+                            .padding()
+                            .background(.ultraThickMaterial)
+                            .cornerRadius(10)
+                            .shadow(radius: 50)
+                    }
```

Comment:
```suggestion
                    if isLoading {
                        ProgressView("Loading...")
                            .padding()
                            .background(.ultraThickMaterial)
                            .cornerRadius(10)
                            .shadow(radius: 50)
                    }
```

Showing the `ProgressView` when the draw status is `inProgress` means it also shows when the user pans around the map, which I think isn't ideal. Instead, consider just using a state boolean and setting it to `true`/`false` when needed.

Reply:
@CalebRas I'm going to implement this change.